src/routes/(app)/person/[id]/+page.svelte

```diff
@@ -31,33 +34,64 @@
     return unsubscribe;
   });
 
-  $: {
-    (async () => {
-      try {
-        person = undefined;
-        pageError = undefined;
-        if (!personId) {
-          return;
-        }
-        const data = await getPerson(personId);
-        person = data;
-      } catch (err: any) {
-        person = undefined;
-        pageError = err;
+  $: if (personId) {
+    fetchPersonData();
+  }
+
+  $: if (sortOption && cachedCredits) {
+    sortedCredits = getSortedCredits(cachedCredits, sortOption);
+  }
+
+  async function fetchPersonData() {
+    try {
+      person = undefined;
+      pageError = undefined;
+      if (!personId) {
+        return;
       }
-    })();
+      person = await getPerson(personId);
+      cachedCredits = await getPersonCredits();
+      sortedCredits = getSortedCredits(cachedCredits, sortOption);
+    } catch (err: any) {
+      person = undefined;
+      pageError = err;
+    }
   }
 
   async function getPerson(id: number) {
     return (await axios.get(`/content/person/${id}`)).data as TMDBPersonDetails;
   }
 
   async function getPersonCredits() {
-    const credits = (await axios.get(`/content/person/${data.personId}/credits`))
+    let credits = (await axios.get(`/content/person/${data.personId}/credits`))
       .data as TMDBPersonCombinedCredits;
-    credits.cast = credits.cast?.sort((a, b) => b.vote_count - a.vote_count);
     return credits;
   }
+
+  function getSortedCredits(credits: TMDBPersonCombinedCredits, sortOption: string) {
+    if (!credits || !credits.cast) return;
+    let sorted = [...credits.cast];
+    switch (sortOption) {
+      case "Vote count":
+        sorted.sort((a, b) => b.vote_count - a.vote_count);
+        break;
+      case "Newest":
```

Comment:
The Newest and Oldest filters seem to be able to break on certain credits lists.

It works on your example page, but for example, can break on [Uma Thurman (/person/139)](http://localhost:5173/person/139).

With her list as an example, if I set the sort to Newest, the list looks right, until I get towards the bottom, where it seems to have just stopped sorting (same thing happens, but opposite with Oldest sort).

It seems to always break after the movie "The Old Guard 2" (in her list specifically), which I don't think has a release date yet, which _might_ be why the sort breaks.